CrazyCanvas/Source/ECS/Systems/Player/HealthSystemClient.cpp

```diff
@@ -106,11 +110,57 @@ bool HealthSystemClient::OnPlayerAliveUpdated(const PlayerAliveUpdatedEvent& eve
 		event.pPlayer->IsDead() ? "true" : "false",
 		playerEntity);
 
+	for (uint32 i = 0; i < m_ParticleEmitters.GetSize(); i++)
+		m_ParticleEmitters.PopBack();
+
 	if (event.pPlayer->IsDead())
 	{
 		ECSCore* pECS = ECSCore::GetInstance();
 		ComponentArray<ChildComponent>* pChildComponents = pECS->GetComponentArray<ChildComponent>();
 
+		PositionComponent positionComponent = pECS->GetComponent<PositionComponent>(playerEntity);
```

Comment:
I think this should be const as well as you're not manipulating the positionComponent